exercises/lens-person/src/Person.hs

```diff
@@ -1,49 +1,43 @@
-module Person where
-
-import           Data.Time.Calendar
-
-data Person = Person {
-      _name    :: Name,
-      _born    :: Born,
-      _address :: Address
-    }
-
-data Name = Name {
-      _foreNames :: String, -- Space separated
-      _surName   :: String
-    }
-
-data Born = Born {
-      _bornAt :: Address,
-      _bornOn :: Day
-    }
-
-data Address = Address {
-      _street      :: String,
-      _houseNumber :: Int,
-      _place       :: String, -- Village / city
-      _country     :: String
-    }
-
--- Valid values of Gregorian are those for which 'Data.Time.Calendar.fromGregorianValid'
--- returns Just.
-data Gregorian = Gregorian {
-      _year  :: Integer,
-      _month :: Int,
-      _day   :: Int
-    }
-
--- Implement these.
+module Person
+  ( Address (..)
+  , Born    (..)
+  , Name    (..)
+  , Person  (..)
+  , bornStreet
+  , renameStreets
+  , setBirthMonth
+  , setCurrentStreet
+  ) where
+
+import Data.Time.Calendar (Day)
+
+data Person = Person { _name    :: Name
+                     , _born    :: Born
+                     , _address :: Address
+                     }
+
+data Name = Name { _foreNames :: String
+                 , _surName   :: String
+                 }
+
+data Born = Born { _bornAt :: Address
+                 , _bornOn :: Day
+                 }
+
+data Address = Address { _street      :: String
+                       , _houseNumber :: Int
+                       , _place       :: String
+                       , _country     :: String
+                       }
 
 bornStreet :: Born -> String
 bornStreet = undefined
 
-setCurrentStreet :: String -> Person -> Person
-setCurrentStreet = undefined
+renameStreets :: (String -> String) -> Person -> Person
```

Comment:
I'm not sure if top level functions should be defined in a solution specific order or if they should be alphabetical.

Alphabetical order usually makes finding things easier, and I usually do the same in the exports list, to match. But I don't have a very strong opinion on this.

I left the data declarations in topological order. I'm inconsistent. :grin: 

Should I revert the functions to their old ordering?